tests/attr/test_deeplift_basic.py

```diff
@@ -78,25 +94,49 @@ def test_relu_deeplift_multi_ref(self):
         baselines = (b1, b2)
 
         model = ReLUDeepLiftModel()
-        self._deeplift_helper(model, DeepLiftShap(model), inputs, baselines)
+        self._deeplift_assert(model, DeepLiftShap(model), inputs, baselines)
 
-    def test_relu_linear_deeplift(self):
+    def test_relu_deepliftshap_baselines_as_function(self):
         model = ReLULinearDeepLiftModel()
-        x1 = torch.tensor([[-10.0, 1.0, -5.0]], requires_grad=True)
-        x2 = torch.tensor([[3.0, 3.0, 1.0]], requires_grad=True)
+        x1 = torch.tensor([[-10.0, 1.0, -5.0]])
+        x2 = torch.tensor([[3.0, 3.0, 1.0]])
 
-        b1 = torch.tensor([[0.0, 0.0, 0.0]], requires_grad=True)
-        b2 = torch.tensor([[0.0, 0.0, 0.0]], requires_grad=True)
+        def gen_baselines():
+            b1 = torch.tensor([[0.0, 0.0, 0.0], [1.0, 1.0, 1.0]])
+            b2 = torch.tensor([[0.0, 0.0, 0.0], [1.0, 1.0, 1.0]])
+            return (b1, b2)
+
+        def gen_baselines_with_inputs(inputs):
+            b1 = inputs[0].mean(0, keepdim=True)
+            b2 = inputs[1].mean(0, keepdim=True)
+            return (b1, b2)
+
+        def gen_baselines_returns_array():
+            b1 = [[0.0, 0.0, 0.0], [1.0, 1.0, 1.0]]
+            b2 = [[0.0, 0.0, 0.0], [1.0, 1.0, 1.0]]
+            return (b1, b2)
 
         inputs = (x1, x2)
-        baselines = (b1, b2)
 
-        # expected = [[[0.0, 0.0]], [[6.0, 2.0]]]
-        self._deeplift_helper(model, DeepLift(model), inputs, baselines)
+        self._deeplift_assert(model, DeepLiftShap(model), inputs, gen_baselines)
+        self._deeplift_assert(
+            model, DeepLiftShap(model), inputs, gen_baselines_with_inputs
+        )
+        with self.assertRaises(AssertionError):
+            self._deeplift_assert(
+                model, DeepLiftShap(model), inputs, gen_baselines_returns_array
+            )
+
+    def _deeplift_assert(self, model, attr_method, inputs, baselines):
+        input_bsz = len(inputs[0])
+        if callable(baselines):
+            baseline_parameters = signature(baselines).parameters
+            if len(baseline_parameters) > 0:
+                baselines = baselines(inputs)
+            else:
+                baselines = baselines()
```

Comment:
nit: Since these tests mostly check deltas and shapes, might also be good to consider adding tests for both methods which confirm that the attributions are the same when either providing the tensor directly or when the same tensor is returned by a callable?